### PR TITLE
Number formatting (almost) everywhere + compact cashout + QoL fixes

### PR DESCRIPTION
**Pull request by stupxd**
Main features:
- Number formatting for tooltip texts, money (current, continue run menu, cashout), poker hand values, and more.
- Compact cashout - hide off-screen cashout rows, which also helps with performance

Small things:
- Fix "Best Hand" number scaling in continue run
image vs image
- Number formatting also fixes floating point error in tooltips.
- Smaller precision for bigger numbers (>e10 only displays 2 digits, >e100 only displays 1 digit)
image image image
- Fix rare floating point error with scientific notation
image
- Fix money text scaling during the run (before, money would keep its scale permanently, and only update when run reloaded)
- Add some better example in testkeybinds

### CODE REVIEW

**Comment by stupxd** [comment]
I think this should be good now

File: core/core.lua [hunk 1]
```diff
@@ -395,6 +395,21 @@ function modsCollectionTally(pool, set)
...
+end
+
+-- Formatting util for UI elements (look number_formatting.toml)
+function format_ui_value(value)
+    if type(value) ~= "number" then
+        return tostring(value)
+    end
+
+    return number_format(value, 1000000)
+end
```
Comment (english5040) [inline core/core.lua:411]: move to utils. though , file structure remains messy

**Comment by english5040** [comment]
Thanks for the PR!

**Comment by english5040** [comment]
ok, looks like scale_number and co. is a ui hack, no idea why they don't use maxw to scale things. whatever, your functions work for now

File: lovely/compact_cashout.toml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+[[patches]]
+[patches.pattern]
+target = "functions/common_events.lua"
+pattern = "if config.name ~= 'bottom' then"
+position = "at"
+payload = '''
+if config.name ~= 'bottom' then
+
+    G.GAME.round_eval_rows = G.GAME.round_eval_rows + 1
```
[the remark this answers is not in the document]
Comment (stupxd) [reply lovely/compact_cashout.toml:21]: Ah, I can change that

File: core/core.lua [hunk 3]
```diff
@@ -395,6 +395,21 @@ function modsCollectionTally(pool, set)
 	return obj_tally
 end
 
+function math.round(num, precision)
+	precision = 10^(precision or 0)
+	
+	return math.floor(num * precision + 0.49999999999999994) / precision
```
[the remark this answers is not in the document]
Comment (stupxd) [reply core/core.lua:401]: > more mathematically precise

Not sure how else would I make that, the only thing I missed is that there's an extra 9 so some numbers do not round properly (e.g. `0.49999999999999999999994` rounds to 1)

File: lovely/compact_cashout.toml (hunk 2 above)
[the remark this answers is not in the document]
Comment (stupxd) [reply lovely/compact_cashout.toml:21]: I guess I'll have to use a global variable, just not in the G.GAME table.

I don't think it's the best idea to override every `add_round_eval_row` call just for this and functions are in different files, so I can't use local value. Feel free to change this if you have a better idea

**Comment by stupxd** [comment]
> ok, looks like scale_number and co. is a ui hack, no idea why they don't use maxw to scale things. whatever, your functions work for now

Yeah, that's very odd, they only use maxw when initializing, the logic is pretty weird afterwards with letters scaling.. couldn't figure that out, this is the only thing that worked. I'm sure there's a better way I just don't have the will to look for it 😭

File: core/core.lua (hunk 3 above)
[the remark this answers is not in the document]
Comment (english5040) [reply core/core.lua]: it's fine for now

**Comment by english5040** [comment]
Alright, I've decided that using a global variable is more effective than manually checking the number of children in the cashout box. merging

**Pull request merged**